internal/scrapertest/compare.go

```diff
@@ -22,9 +22,18 @@ import (
 	"go.uber.org/multierr"
 )
 
+// CompareOption is applied by the CompareMetricSlices function
+// to mutates an expected and/or actual result before comparing
+type CompareOption func(expected, actual pdata.MetricSlice)
```

Comment:
Is the intent to allow users to write custom "options"?

Reply:
Not necessarily, but I could image it being useful once in a while (e.g. some completely untenable metric just needs to be scrubbed and tested elsewhere.) I'll tweak the pattern though, cause it's not necessary now.

Reply:
Now only a `CompareOption` interface w/ `apply` method is being exported, and implementation is provided via `func IgnoreValues() CompareOption { return ignoreValues{} }`.